crates/rpc/src/cairo/starknet_rs/pending.rs

```diff
@@ -0,0 +1,63 @@
+use std::collections::HashMap;
+
+use pathfinder_common::state_update::ContractUpdate;
+use pathfinder_common::StateUpdate;
+use starknet_in_rust::core::errors::state_errors::StateError;
+use starknet_in_rust::felt::Felt252;
+use starknet_in_rust::utils::Address;
+
+pub(crate) fn apply_pending_update<S: starknet_in_rust::state::state_api::State>(
+    state: &mut S,
+    pending_update: &StateUpdate,
+) -> Result<(), StateError> {
+    // NOTE: class _declarations_ are handled during sync. We download and insert new class declarations for the pending block
+    // after downloading it -- here we build on the fact that those are already available in the database -- and thus in the state
+    // as well...
+
+    let mut address_to_class_hash: HashMap<Address, starknet_in_rust::utils::ClassHash> =
+        Default::default();
+    let mut address_to_nonce: HashMap<Address, Felt252> = Default::default();
+    let mut storage_updates: HashMap<Address, HashMap<Felt252, Felt252>> = Default::default();
+
+    for (
+        contract_address,
+        ContractUpdate {
+            storage,
+            class,
+            nonce,
+        },
+    ) in &pending_update.contract_updates
```

Comment:
We're missing the system contract updates :P I think? If they are even present in the pending diff..